src/components/SearchPanel.vue

```diff
@@ -4,112 +4,552 @@
+     */
+    getLabelOperator(operator) {
+      return `[ ${FILTER_EXPRESSION_OPERATORS[operator]} ]`
+    },
+
+    isSelectDisabled(input) {
+      return this.state.loading[input.dependance] || input.loading || input.disabled;
+    },
+
+    async onFocus(e) {
       if (this.isMobile()) {
-        const top = $(event.target).position().top - 10 ;
+        const top = $(e.target).position().top - 10 ;
         await this.$nextTick();
         setTimeout(() => $('.sidebar').scrollTop(top), 500);
       }
     },
-    async autocompleteRequest(params={}){
-      return this.$options.service.getUniqueValuesFromField({
-        ...params,
-        output: 'autocomplete'
-      });
+
+    /**
+     * Sync `this.state.forminputs` with `input.value`
+     */
+    async changeInput(input) {
+      console.log(input);
+      const field       = input.attribute;
+      const deps        = this.state.forminputs.filter(d => d.dependance === field);  // get inputs that depends on the current one
+      const state       = this.state;
+      let value         = input.value;
+
+      console.log(input, deps);
+
+      const is_invalid       = v => [SEARCH_ALLVALUE, null, undefined].includes(v) || '' === v.toString().trim(); // whether father input can search on subscribers
+      const is_autocomplete  = s => 'autocompletefield' === s.type;
+      const has_dependance   = s => ['selectfield', 'autocompletefield'].includes(s.type) && !s.dependance_strict && s.dependance
+      const is_valuemap      = s => !!(has_dependance(s) && s.values.length);
+      const is_valuerelation = s => !!(has_dependance(s) && !s.values.length && s.options.layer_id);
+
+      try {
+        this.state.searching = true;
+
+        if ('numberfield' === input.type) {
+          value = value || 0 === value ? value : null;
+        }
+
+        // fallback to default value → `SEARCH_ALLVALUE`
+        if (undefined === value) {
+          value = SEARCH_ALLVALUE;
+        }
+
+        /** @TODO check if it has one reason to trim  */
+        if (!['textfield', 'textField'].includes(input.type)) {
+          value = value.trim();
+        }
+
+        input.value = value;
+
+        // skip when no dependants
+        if (!deps.length) {
+          console.info('no deps for: ', input)
+          return;
+        }
+
+        // loop over dependants
+        deps.forEach(s => {
+
+          // autocomplete    → reset values to empty array
+          // no autocomplete → get first value (ALL_VALUE)
+          s.values.splice(is_autocomplete(s) || is_invalid(value) ? 0 : 1);
+
+          // parent has invalid value (eg. ALL_VALUE) → set all values to subscribe 
+          if (is_invalid(value) && !is_autocomplete(s)) {
+            setTimeout(() => s.values = [...s._values]);
+          }
+
+          s.value = 'selectfield' === s.type ? SEARCH_ALLVALUE : null;
+        });
+
+        // value is empty → disable dependants inputs
+        if (!value || value === SEARCH_ALLVALUE) {
```

Comment:
is this correct?

```suggestion
        if (is_invalid(value)) {
```

ie. the original version also excludes zero